kits/cpp/simple/lux/city.hpp

```diff
@@ -52,22 +52,14 @@ namespace lux
         string cityid;
         int team;
         float fuel;
-        vector<CityTile *> citytiles;
+        vector<CityTile> citytiles{};
         float lightUpkeep;
         City(){};
-        City(int teamid, const string &cityid, float fuel, float lightUpkeep)
-        {
-            this->cityid = cityid;
-            this->team = teamid;
-            this->fuel = fuel;
-            this->citytiles = vector<CityTile *>();
-            this->lightUpkeep = lightUpkeep;
-        }
+        City(int teamid, const string &cityid, float fuel, float lightUpkeep) : cityid(cityid), team(teamid), fuel(fuel), lightUpkeep(lightUpkeep) {}
         CityTile* addCityTile(int x, int y, int cooldown)
         {
-            CityTile* ct = new CityTile(this->team, this->cityid, x, y, cooldown);
-            this->citytiles.push_back(ct);
-            return ct;
+            citytiles.emplace_back(team, cityid, x, y, cooldown);
+            return &citytiles.back();
```

Comment:
Whoops, I noticed a mistake here. This pointer becomes invalid as soon as the vector `citytiles` resizes itself (e.g., if `citytiles` starts out with capacity 1, and then you add a second city tile). I'll fix it momentarily.